server/nri-api.go

```diff
@@ -0,0 +1,741 @@
+		),
+		nrigen.WithResourceChecker(
+			func(r *rspec.LinuxResources) error {
+				if r == nil {
+					return nil
+				}
+				if mem := r.Memory; mem != nil {
+					if mem.Limit != nil {
+						if err := cgmgr.VerifyMemoryIsEnough(*mem.Limit); err != nil {
+							return err
+						}
+					}
+					if !node.CgroupHasMemorySwap() {
+						mem.Swap = nil
+					}
+				}
+				if !node.CgroupHasHugetlb() {
+					r.HugepageLimits = nil
+				}
+				return nil
+			},
+		),
+		nrigen.WithBlockIOResolver(
+			func(className string) (*rspec.LinuxBlockIO, error) {
+				if !a.cri.Config().BlockIO().Enabled() || className == "" {
+					return nil, nil
+				}
+				if blockIO, err := blockio.OciLinuxBlockIO(className); err == nil {
+					return blockIO, nil
+				}
+				return nil, nil
+			},
+		),
+		nrigen.WithRdtResolver(
+			func(className string) (*rspec.LinuxIntelRdt, error) {
+				if className == "" || className == "/PodQos" {
+					return nil, nil
+				}
+				return &rspec.LinuxIntelRdt{
+					ClosID: rdt.ResctrlPrefix + className,
+				}, nil
+			},
+		),
+	)
+
+	if err := wrapgen.Adjust(adjust); err != nil {
+		return fmt.Errorf("failed to adjust container %s: %w", ctr.GetID(), err)
+	}
+
+	return nil
+}
+
+func (a *nriAPI) postCreateContainer(ctx context.Context, criPod *sandbox.Sandbox, criCtr *oci.Container) error {
+	if !a.isEnabled() {
+		return nil
+	}
+
+	pod := nriPodSandbox(criPod)
+	ctr := &criContainer{
+		api: a,
+		ctr: criCtr,
+	}
+
+	err := a.nri.PostCreateContainer(ctx, pod, ctr)
+
+	return err
+}
+
+func (a *nriAPI) startContainer(ctx context.Context, criPod *sandbox.Sandbox, criCtr *oci.Container) error {
+	if !a.isEnabled() {
+		return nil
+	}
+
+	pod := nriPodSandbox(criPod)
+	ctr := &criContainer{
+		api: a,
+		ctr: criCtr,
+	}
+
+	err := a.nri.StartContainer(ctx, pod, ctr)
+
+	return err
+}
+
+func (a *nriAPI) postStartContainer(ctx context.Context, criPod *sandbox.Sandbox, criCtr *oci.Container) error {
+	if !a.isEnabled() {
+		return nil
+	}
+
+	pod := nriPodSandbox(criPod)
+	ctr := &criContainer{
+		api: a,
+		ctr: criCtr,
+	}
+
+	err := a.nri.PostStartContainer(ctx, pod, ctr)
+
+	return err
+}
+
+func (a *nriAPI) updateContainer(ctx context.Context, criCtr *oci.Container, req *cri.LinuxContainerResources) (*cri.LinuxContainerResources, error) {
+	if !a.isEnabled() {
+		return req, nil
+	}
+
+	const noOomAdj = 0
+
+	criPod := a.cri.getSandbox(ctx, criCtr.Sandbox())
+	pod := nriPodSandbox(criPod)
+	ctr := &criContainer{
+		api: a,
+		ctr: criCtr,
+	}
+
+	r, err := a.nri.UpdateContainer(ctx, pod, ctr, api.FromCRILinuxResources(req))
+	if err != nil {
+		return nil, err
+	}
+
+	return r.ToCRI(noOomAdj), nil
+}
+
+func (a *nriAPI) postUpdateContainer(ctx context.Context, criCtr *oci.Container) error {
+	if !a.isEnabled() {
+		return nil
+	}
+
+	criPod := a.cri.getSandbox(ctx, criCtr.Sandbox())
+	pod := nriPodSandbox(criPod)
+	ctr := &criContainer{
+		api: a,
+		ctr: criCtr,
+	}
+
+	err := a.nri.PostUpdateContainer(ctx, pod, ctr)
```

Comment:
same here, can you go through and compress the lines where we do `err := ...; return err` and just do `return ...`